### PR TITLE
fix: resolve build errors

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: packages/result/src/lib/Option/None.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+import type { Option, Some } from '../Option';
 import { createErr, type ResultErr } from '../Result/Err';
 import { createOk, type ResultOk } from '../Result/Ok';
 import type { IOption } from './IOption';
 import { OptionError } from './OptionError';
 import type { OptionSome } from './Some';
 
 export class OptionNone implements IOption<any> {
-	public isSome(): false {
+	public isSome(): this is Some<any> & false {
```
Comment (vladfrangu) [inline packages/result/src/lib/Option/None.ts:10]: Remove the `& false`, its useless

File: packages/result/src/lib/Option/None.ts [hunk 2]
```diff
@@ -108,7 +108,7 @@ export class OptionNone implements IOption<any> {
 	}
 
 	public filter(predicate: (value: never) => boolean): OptionNone;
```
Comment (vladfrangu) [inline packages/result/src/lib/Option/None.ts:110]: As well as all other places

```suggestion
	public filter(predicate: (value: never) => boolean): this;
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/result/src/lib/Option/None.ts [hunk 3]
```diff
@@ -7,11 +7,11 @@ import { OptionError } from './OptionError';
...
 
 export class OptionNone implements IOption<any> {
-	public isSome(): false {
+	public isSome(): this is OptionSome<any> {
 		return false;
 	}
 
-	public isSomeAnd(cb?: (value: never) => boolean): false;
+	public isSomeAnd(cb?: (value: never) => boolean): this is OptionSome<any>;
 	public isSomeAnd(): false {
```
Comment (vladfrangu) [inline packages/result/src/lib/Option/None.ts:15]: This should also be a typeguard, no?

File: packages/iterator-utilities/package.json [hunk 4]
```diff
@@ -687,7 +687,7 @@
 		}
 	},
 	"sideEffects": [
-		"./dist/esm/chunk-G5GHKT7C.mjs",
+		"./dist/esm/chunk-PAWJFY3S.mjs",
```
Comment (favna) [inline packages/iterator-utilities/package.json:690]: revert

**Review by favna — CHANGES_REQUESTED** [review]

File: packages/result/src/lib/Option/None.ts (hunk 3 above)
Comment (kyranet) [reply packages/result/src/lib/Option/None.ts:15]: Unneeded, this method isn't overloaded.

**Review by vladfrangu — APPROVED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**